cache/disk/disk_test.go

```diff
@@ -183,6 +184,44 @@ func TestCachePutWrongSize(t *testing.T) {
 	}
 }
 
+func TestCacheGetContainsWrongSize(t *testing.T) {
+	cacheDir := tempDir(t)
+	defer os.RemoveAll(cacheDir)
+	testCache := New(cacheDir, 100, nil)
+
+	content := "hello"
+	hash := hashStr(content)
+	correctSize := int64(len(content))
+
+	var err error
+	var found bool
+
+	err = testCache.Put(cache.CAS, hash, correctSize, strings.NewReader(content))
+	if err != nil {
+		t.Fatal("Expected success", err)
+	}
+
+	found, _ = testCache.Contains(cache.CAS, contentsHash, correctSize+1)
+	if found {
+		t.Error("Expected not found, due to size being different")
+	}
+
+	_, _, err = testCache.Get(cache.CAS, contentsHash, correctSize+1)
+	if err == nil {
+		t.Error("Expected error, due to size being different")
+	}
```

Comment:
I think this should just be a cache miss, no error.

Reply:
Done